File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/delete/RecordDeleteDao.java
package ru.citeck.ecos.records3.record.op.delete;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.records3.record.dao.RecordsDao;

public interface RecordDeleteDao extends RecordsDao {

    @NotNull
    DelStatus delete(@NotNull String recordId);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/delete/RecordDeleteDao.kt
package ru.citeck.ecos.records3.record.dao.delete

import ru.citeck.ecos.records3.record.dao.RecordsDao
import kotlin.jvm.Throws

interface RecordDeleteDao : RecordsDao {

    @Throws(Exception::class)
    fun delete(recordId: String): DelStatus
}
